Pict-rs: add module + package fixes

happysalada opened the pull request:
###### Motivation for this change

Just adding a module to enable deployment

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

happysalada committed:
pict-rs: fixes

billewanick commented:
Result of `nixpkgs-review pr 139868` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pict-rs</li>
  </ul>
</details>

aanderse commented on nixos/modules/services/web-apps/pict-rs.nix:
```diff
@@ -0,0 +1,45 @@
+{
+  options.services.pict-rs = {
+    enable = mkEnableOption "pict-rs server";
+    imageDirectory = mkOption {
```
Personally I would call this `stateDir` or `dataDir` for consistency, but that's more opinion than anything. For this application the images are state.

happysalada replied:
happy to change this. I have to say I was looking for a good name. I'm going to go for dataDir since it's a bit more expressive.

aanderse commented on nixos/modules/services/web-apps/pict-rs.nix:
```diff
@@ -0,0 +1,45 @@
+in
+{
+  options.services.pict-rs = {
+    enable = mkEnableOption "pict-rs server";
```
Bonus points of you mention something about throwing this begging a reverse proxy and include upstream documentation on how to do so

aanderse commented on nixos/modules/services/web-apps/pict-rs.nix:
```diff
@@ -0,0 +1,45 @@
+      '';
+    };
+  };
+  config = lib.mkIf cfg.enable {
```
Even more bonus points of this gets a NixOS test. I think a test would be helpful.

happysalada replied:
you're right!

aanderse reviewed (approved):
I love it. I also love my opinions, expressed as a review, but understand if you don't.

:smile:

SuperSandro2000 reviewed (approved):
Thanks for cleaning up the package, gets a LGTM from me

happysalada committed:
pict-rs: add initial module

happysalada commented:
Added a very basic test, some documentation and changed the variable name.
I'm going to merge this for now, but there will probably be a follow up PR (all the functionalities will only really be tested with lemmy). I'm happy to take care of anything that you think should change in the follow up PR
